src/nvim/tui/tui.c

```diff
@@ -1230,6 +1228,17 @@ static void show_termcap_event(void **argv)
   verbose_stop();  // flush now
 }
 
+/// Ask the terminal to send us the background color.
+static void ask_for_terminal_bg(UI *ui)
+{
+  if (!option_was_set("bg") && !tui_is_stopped(ui)) {
```

Comment:
Please do not call the function that accesses the global variables of the main thread from the TUI thread.

https://github.com/neovim/neovim/blob/111d04fde64f8be8409cd3b1b96886aee2ac8ff5/src/nvim/tui/tui.c#L425

Accessing the global variables of the main thread from the TUI thread after this `CONTINUE(bridge);` may result in a data race.